api/client/copy.go

```diff
@@ -0,0 +1,218 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/pkg/archive"
+)
+
+type copyDirection int
+
+const (
+	fromContainer copyDirection = (1 << iota)
+	toContainer
+	acrossContainers = fromContainer | toContainer
+)
+
+// CmdCp handles copying files or directories to/from/across containers.
+func (cli *DockerCli) CmdCp(args ...string) error {
+	cmd := cli.Subcmd("cp", "[SRC_CONTAINER:]SRC_PATH [DST_CONTAINER:]DST_PATH", "Copy files between containers or a local path", true)
```

Comment:
Will Windows Paths also use slashes, or back-slashes? i.e. a regular path on Windows will be `C:\some\host\path`. But, yes, this will become an issue if Windows *containers* are going to be supported I guess (I suspect that will still be far in the future)